Documentation for ENG-37662 Dry-run Drift Detection (#103)
NSO-developer/nso-gitbook: operation-and-usage/cli/introduction-to-nso-cli.md

<details>

<summary><code>dry-run-drift-detection (true | false)</code></summary>

> When enabled, the user will, when performing a commit, be informed if the changeset of the transaction has changed since the user last did a dry-run, to avoid committing unintended changes. This setting exists in "warn" and "strict" mode, see below.

Reviewer: ```suggestion
When enabled, the user is notified during commit if the transaction’s changeset has changed since their last dry-run, helping prevent unintended changes from being committed. This setting is available in `warn` and `strict` modes; see below.
```

</details>

<details>

<summary><code>dry-run-drift-detection-mode (warn | strict)</code></summary>

> This is the mode for the above dry-run-drift-detection setting. In "warn" mode, the CLI will return a warning if a changeset mismatch is detected between dry-run and commit, prompting the user on whether or not to proceed anyway. In "strict" mode, the commit will be aborted, and the user needs to perform a new dry-run in order to commit.

Reviewer 2: ```suggestion
This is the mode for the above dry-run-drift-detection setting. In "warn" mode, the CLI will return a warning if a changeset mismatch is detected between dry-run and commit, asking the user whether to proceed. In "strict" mode, the commit will be aborted, and the user needs to perform a new dry-run in order to commit.
```

> This is the mode for the above dry-run-drift-detection setting. In "warn" mode, the CLI will return a warning if a changeset mismatch is detected between dry-run and commit, prompting the user on whether or not to proceed anyway. In "strict" mode, the commit will be aborted, and the user needs to perform a new dry-run in order to commit.

Reviewer: ```suggestion
This is the mode for the `dry-run-drift-detection` setting. In `warn` mode, the CLI warns if the changeset differs between dry-run and commit, and prompts the user whether to proceed. In `strict` mode, the commit is aborted, and the user must perform a new dry-run before committing.
```
